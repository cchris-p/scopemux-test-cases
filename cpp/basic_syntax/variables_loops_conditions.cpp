/**
 * @file variables_loops_conditions.cpp
 * @brief Demonstration of variables, loops, and conditional statements in C++
 *
 * This example shows:
 * - Various variable declarations and types including auto
 * - for, range-based for, while, and do-while loops
 * - if, else if, else conditions
 * - switch statements
 */

#include <iostream>
#include <vector>
#include <string>

/**
 * @brief Program entry point
 * @return Exit status code
 */
int main() {
    // Variable declarations
    int i = 0;
    float f = 3.14f;
    double d = 2.71828;
    char c = 'A';
    bool b = true;
    std::string s = "Hello, C++";
    auto a = 42;  // Type inference
    
    // Standard array
    int array[5] = {1, 2, 3, 4, 5};
    
    // STL container
    std::vector<int> vec = {10, 20, 30, 40, 50};
    
    // Standard for loop
    std::cout << "Standard for loop example:" << std::endl;
    for (i = 0; i < 5; i++) {
        std::cout << "array[" << i << "] = " << array[i] << std::endl;
    }
    
    // Range-based for loop (C++11)
    std::cout << "\nRange-based for loop example:" << std::endl;
    for (const auto& value : vec) {
        std::cout << "value: " << value << std::endl;
    }
    
    // While loop
    std::cout << "\nWhile loop example:" << std::endl;
    i = 0;
    while (i < 5) {
        std::cout << "iteration " << i << std::endl;
        i++;
    }
    
    // Do-while loop
    std::cout << "\nDo-while loop example:" << std::endl;
    i = 0;
    do {
        std::cout << "iteration " << i << std::endl;
        i++;
    } while (i < 5);
    
    // If-else statements
    std::cout << "\nIf-else example:" << std::endl;
    if (b) {
        std::cout << "b is true" << std::endl;
    } else {
        std::cout << "b is false" << std::endl;
    }
    
    // If-else if-else
    std::cout << "\nIf-else if-else example:" << std::endl;
    if (i < 3) {
        std::cout << "i is less than 3" << std::endl;
    } else if (i == 5) {
        std::cout << "i is equal to 5" << std::endl;
    } else {
        std::cout << "i is greater than 5" << std::endl;
    }
    
    // Switch statement
    std::cout << "\nSwitch example:" << std::endl;
    switch (c) {
        case 'A':
            std::cout << "c is 'A'" << std::endl;
            break;
        case 'B':
            std::cout << "c is 'B'" << std::endl;
            break;
        default:
            std::cout << "c is not 'A' or 'B'" << std::endl;
            break;
    }
    
    return 0;
}
