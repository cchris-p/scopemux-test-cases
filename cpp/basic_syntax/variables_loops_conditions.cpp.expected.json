{
  "language": "C++",
  "ast": {
    "type": "translation_unit",
    "functions": [
      {
        "name": "main",
        "type": "function_definition",
        "return_type": "int",
        "parameters": [],
        "body_range": {
          "start_line": 18,
          "start_column": 0,
          "end_line": 92,
          "end_column": 1
        },
        "docstring": "Program entry point\nReturn Exit status code"
      }
    ],
    "includes": [
      {
        "path": "iostream",
        "system": true,
        "line": 11
      },
      {
        "path": "vector",
        "system": true,
        "line": 12
      },
      {
        "path": "string",
        "system": true,
        "line": 13
      }
    ],
    "comments": [
      {
        "type": "block_comment",
        "text": "This example shows:\n- Various variable declarations and types including auto\n- for, range-based for, while, and do-while loops\n- if, else if, else conditions\n- switch statements",
        "range": {
          "start_line": 1,
          "start_column": 0,
          "end_line": 9,
          "end_column": 3
        }
      },
      {
        "type": "block_comment",
        "text": "Program entry point\nReturn Exit status code",
        "range": {
          "start_line": 15,
          "start_column": 0,
          "end_line": 18,
          "end_column": 3
        }
      },
      {
        "type": "line_comment",
        "text": "Variable declarations",
        "line": 19
      },
      {
        "type": "line_comment",
        "text": "Type inference",
        "line": 25
      },
      {
        "type": "line_comment",
        "text": "Standard array",
        "line": 27
      },
      {
        "type": "line_comment",
        "text": "STL container",
        "line": 30
      },
      {
        "type": "line_comment",
        "text": "Standard for loop",
        "line": 33
      },
      {
        "type": "line_comment",
        "text": "Range-based for loop (C++11)",
        "line": 39
      },
      {
        "type": "line_comment",
        "text": "While loop",
        "line": 45
      },
      {
        "type": "line_comment",
        "text": "Do-while loop",
        "line": 53
      },
      {
        "type": "line_comment",
        "text": "If-else statements",
        "line": 62
      },
      {
        "type": "line_comment",
        "text": "If-else if-else",
        "line": 71
      },
      {
        "type": "line_comment",
        "text": "Switch statement",
        "line": 82
      }
    ],
    "variable_declarations": [
      {
        "name": "i",
        "type": "int",
        "line": 19,
        "initialized": true
      },
      {
        "name": "f",
        "type": "float",
        "line": 20,
        "initialized": true
      },
      {
        "name": "d",
        "type": "double",
        "line": 21,
        "initialized": true
      },
      {
        "name": "c",
        "type": "char",
        "line": 22,
        "initialized": true
      },
      {
        "name": "b",
        "type": "bool",
        "line": 23,
        "initialized": true
      },
      {
        "name": "s",
        "type": "std::string",
        "line": 24,
        "initialized": true
      },
      {
        "name": "a",
        "type": "auto",
        "line": 25,
        "initialized": true
      },
      {
        "name": "array",
        "type": "int[5]",
        "line": 28,
        "initialized": true
      },
      {
        "name": "vec",
        "type": "std::vector<int>",
        "line": 31,
        "initialized": true
      }
    ],
    "loops": [
      {
        "type": "for_statement",
        "range": {
          "start_line": 35,
          "start_column": 4,
          "end_line": 37,
          "end_column": 5
        }
      },
      {
        "type": "range_for_statement",
        "range": {
          "start_line": 41,
          "start_column": 4,
          "end_line": 43,
          "end_column": 5
        }
      },
      {
        "type": "while_statement",
        "range": {
          "start_line": 48,
          "start_column": 4,
          "end_line": 51,
          "end_column": 5
        }
      },
      {
        "type": "do_statement",
        "range": {
          "start_line": 56,
          "start_column": 4,
          "end_line": 59,
          "end_column": 20
        }
      }
    ],
    "conditionals": [
      {
        "type": "if_statement",
        "range": {
          "start_line": 64,
          "start_column": 4,
          "end_line": 68,
          "end_column": 5
        }
      },
      {
        "type": "if_statement",
        "range": {
          "start_line": 73,
          "start_column": 4,
          "end_line": 79,
          "end_column": 5
        }
      },
      {
        "type": "switch_statement",
        "range": {
          "start_line": 84,
          "start_column": 4,
          "end_line": 95,
          "end_column": 5
        }
      }
    ],
    "namespace_references": [
      {
        "name": "std",
        "line": 24,
        "context": "std::string"
      },
      {
        "name": "std",
        "line": 31,
        "context": "std::vector<int>"
      },
      {
        "name": "std",
        "line": 34,
        "context": "std::cout"
      },
      {
        "name": "std",
        "line": 34,
        "context": "std::endl"
      }
    ]
  }
}
